v23.1/architecture/transaction-layer.md

```diff
@@ -198,8 +198,14 @@ The manager accommodates this by allowing transactional requests to acquire lock
 
 However, not all locks are stored directly under the manager's control, so not all locks are discoverable during sequencing. Specifically, write intents (replicated, exclusive locks) are stored inline in the MVCC keyspace, so they are not detectable until request evaluation time. To accommodate this form of lock storage, the manager integrates information about external locks with the concurrency manager structure.
 
+<a name="unreplicated-locks"></a>
+
 {{site.data.alerts.callout_info}}
-Currently, the concurrency manager operates on an unreplicated lock table structure. In the future, we intend to pull all locks, including those associated with [write intents](#write-intents), into the concurrency manager directly through a replicated lock table structure.
+Currently, the concurrency manager operates on an unreplicated lock table structure. Unreplicated locks are held only on a single replica in a [range](overview.html#architecture-range), which is typically the [leaseholder](replication-layer.html#leases). They are very fast to acquire and release, but provide no guarantee of survivability across [lease transfers or leaseholder crashes](replication-layer.html#how-leases-are-transferred-from-a-dead-node).
+
+This lack of survivability for unreplicated locks affects SQL statements implemented using them, such as [`SELECT ... FOR UPDATE`](../select-for-update.html#known-limitations).
+
+In the future, we intend to pull all locks, including those associated with [write intents](#write-intents), into the concurrency manager directly through a replicated lock table structure.
```

Comment:
```suggestion
In the future, we intend to pull all locks, including those associated with [write intents](#write-intents), into the concurrency manager directly through a replicated lock table structure. For details, see [cockroachdb/cockroach#75456](https://github.com/cockroachdb/cockroach/issues/75456).
```

Reply:
I'm not sure if it's worth explaining in external documentation, but the concurrency manager is already how we deal with conflicts on both replicated and unreplicated locks.  It's just that it's less important to keep track of the replicated locks in memory, because if we don't see the lock and optimistically try to go ahead with our operation, it will end up encountering the write intent on disk (and then go back to the concurrency manager to deal with it).

All this is to say, not tracking replicated locks in memory is not a limitation.

Reply:
the line

> In the future, we intend to pull all locks, including those associated with write intents, into the lock table directly.

was added in https://github.com/cockroachdb/docs/pull/7020

do you think we should remove it?  It's probably not really "compliant with policy" for our documentation to make such forward-facing statements anyway, unless talking about a specific known limitation (which you have just informed me this is not)